Moya/Endpoint.swift

```diff
@@ -17,17 +19,23 @@ public enum EndpointSampleResponse {
 }
 
 
-/// Class for reifying a target of the T enum unto a concrete Endpoint
-public class Endpoint<T> {
+/// Class for reifying a target of the Target enum unto a concrete Endpoint.
+public class Endpoint<Target> {
     public let URL: String
     public let method: Moya.Method
     public let sampleResponse: EndpointSampleResponse
-    public let parameters: [String: AnyObject]
+    public let parameters: [String: AnyObject]?
     public let parameterEncoding: Moya.ParameterEncoding
     public let httpHeaderFields: [String: String]
 
     /// Main initializer for Endpoint.
-    public init(URL: String, sampleResponse: EndpointSampleResponse, method: Moya.Method = Moya.Method.GET, parameters: [String: AnyObject] = [String: AnyObject](), parameterEncoding: Moya.ParameterEncoding = .URL, httpHeaderFields: [String: String] = [String: String]()) {
+    public init(URL: String,
+        sampleResponse: EndpointSampleResponse,
+        method: Moya.Method = Moya.Method.GET,
+        parameters: [String: AnyObject]? = nil,
+        parameterEncoding: Moya.ParameterEncoding = .URL,
+        httpHeaderFields: [String: String] = [String: String]()) {
+
```

Comment:
Parameters is optional, but headers are not, that caught my eye.  Should they be the same?